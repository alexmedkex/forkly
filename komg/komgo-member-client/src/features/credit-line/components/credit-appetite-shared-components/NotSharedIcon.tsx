import * as React from 'react'
import { ReactComponent as CloseIcon } from '../../../../styles/themes/komgo/assets/fonts/close.svg'
export const NotSharedIcon = (props: any) => {
  return <CloseIcon data-test-id="not-shared-icon" {...props} />
}
