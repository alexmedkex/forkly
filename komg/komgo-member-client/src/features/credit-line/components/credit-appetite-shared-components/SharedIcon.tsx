import * as React from 'react'
import { ReactComponent as CheckIcon } from '../../../../styles/themes/komgo/assets/fonts/check-green.svg'
export const SharedIcon = (props: any) => {
  return <CheckIcon data-test-id="shared-icon" {...props} />
}
