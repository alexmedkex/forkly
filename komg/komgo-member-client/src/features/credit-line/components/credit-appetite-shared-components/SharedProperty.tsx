import React, { Fragment } from 'react'
import styled from 'styled-components'
import { blueGrey } from '../../../../styles/colors'
import { SharedIcon } from './SharedIcon'
import { NotSharedIcon } from './NotSharedIcon'
import { toKebabCase } from '../../../../utils/casings'

export interface IProps {
  shared: boolean
  label: string
  value?: string
}

export default class SharedProperty extends React.Component<IProps> {
  renderInfo() {
    return (
      <span
        data-test-id={`${toKebabCase(this.props.label)}-label`}
        style={!this.props.shared ? { color: blueGrey } : {}}
      >
        {this.props.label}
        {this.props.value ? <span> - {this.props.value}</span> : null}
      </span>
    )
  }

  render() {
    return (
      <Wrapper>
        {this.props.shared ? (
          <Fragment>
            <SharedIcon />
            {this.renderInfo()}
          </Fragment>
        ) : (
          <Fragment>
            <NotSharedIcon />
            {this.renderInfo()}
          </Fragment>
        )}
      </Wrapper>
    )
  }
}

const Wrapper = styled.div`
  display: inline-block span {
    vertical-align: middle;
  }

  svg {
    margin-right: 10px;
    width: 14px;
    height: 14px;
    vertical-align: middle;
  }
`
