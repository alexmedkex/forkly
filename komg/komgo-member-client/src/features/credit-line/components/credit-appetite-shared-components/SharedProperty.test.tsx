import * as React from 'react'
import renderer from 'react-test-renderer'
import SharedProperty, { IProps } from './SharedProperty'
import { shallow } from 'enzyme'
import { SharedIcon } from './SharedIcon'
import { NotSharedIcon } from './NotSharedIcon'

describe('SharedProperty', () => {
  const props: IProps = {
    shared: false,
    label: 'data',
    value: '1 days'
  }

  const sharedProps = {
    ...props,
    shared: true
  }

  it('should match snapshot', () => {
    expect(renderer.create(<SharedProperty {...props} />).toJSON()).toMatchSnapshot()
  })

  it('should display shared icon if shared', () => {
    const wrapper = shallow(<SharedProperty {...sharedProps} />)
    expect(wrapper.find(SharedIcon).exists()).toBeTruthy()
  })

  it('should display not shared icon if not shared', () => {
    const wrapper = shallow(<SharedProperty {...props} />)
    expect(wrapper.find(NotSharedIcon).exists()).toBeTruthy()
  })

  it('should display label', () => {
    const propsWithoutLabel = {
      ...props,
      value: undefined
    }
    const wrapper = shallow(<SharedProperty {...propsWithoutLabel} />)
    expect(
      wrapper
        .find('span')
        .first()
        .text()
    ).toBe(props.label)
  })

  it('should display label with additional data', () => {
    const propsWithoutLabel = {
      ...props
    }
    const wrapper = shallow(<SharedProperty {...propsWithoutLabel} />)
    const label = wrapper.find('span').first()
    expect(
      label
        .children()
        .find('span')
        .text()
    ).toBe(' - ' + props.value)
  })
})
