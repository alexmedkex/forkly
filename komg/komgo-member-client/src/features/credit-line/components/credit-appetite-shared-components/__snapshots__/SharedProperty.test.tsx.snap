// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedProperty should match snapshot 1`] = `
.c0 display:inline-block span {
  vertical-align: middle;
}

.c0 svg {
  margin-right: 10px;
  width: 14px;
  height: 14px;
  vertical-align: middle;
}

<div
  className="c0"
>
  <svg
    data-test-id="not-shared-icon"
  >
    close.svg
  </svg>
  <span
    data-test-id="data-label"
    style={
      Object {
        "color": "#5d768f",
      }
    }
  >
    data
    <span>
       - 
      1 days
    </span>
  </span>
</div>
`;
